discovery-infra/test_infra/helper_classes/kube_helpers.py

```diff
@@ -0,0 +1,495 @@
+        )
+
+    def get(self) -> dict:
+        return self.crd_api.get_namespaced_custom_object(
+            group=self._hive_api_group,
+            version='v1',
+            plural=self._plural,
+            name=self.ref.name,
+            namespace=self.ref.namespace
+        )
+
+    def delete(self) -> None:
+        self.crd_api.delete_namespaced_custom_object(
+            group=self._hive_api_group,
+            version='v1',
+            plural=self._plural,
+            name=self.ref.name,
+            namespace=self.ref.namespace
+        )
+
+        logger.info('deleted cluster deployment %s', self.ref)
+
+    def status(
+            self,
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATUS_TIMEOUT
+    ) -> dict:
+        """
+        Status is a section in the CRD that is created after registration to
+        assisted service and it defines the observed state of ClusterDeployment.
+        Since the status key is created only after resource is processed by the
+        controller in the service, it might take a few seconds before appears.
+        """
+        def _attempt_to_get_status() -> dict:
+            return self.get()['status']
+
+        return waiting.wait(
+            _attempt_to_get_status,
+            sleep_seconds=0.5,
+            timeout_seconds=timeout,
+            waiting_for=f'cluster {self.ref} status',
+            expected_exceptions=KeyError
+        )
+
+    def state(
+            self,
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATE_TIMEOUT
+    ) -> Tuple[str, str]:
+        state, state_info = None, None
+        for condition in self.status(timeout).get('conditions', []):
+            reason = condition.get('reason')
+
+            if reason == 'AgentPlatformState':
+                state = condition.get('message')
+            elif reason == 'AgentPlatformStateInfo':
+                state_info = condition.get('message')
+
+            if state and state_info:
+                break
+
+        return state, state_info
+
+    def wait_for_state(
+            self,
+            required_state: str,
+            timeout: Union[int, float] = DEFAULT_WAIT_FOR_CRD_STATE_TIMEOUT
+    ) -> None:
+        required_state = required_state.lower()
+
+        def _has_required_state() -> bool:
+            state, _ = self.state(timeout=0.5)
+            return state.lower() == required_state
+
+        waiting.wait(
+            _has_required_state,
+            timeout_seconds=timeout,
+            waiting_for=f'cluster {self.ref} state to be {required_state}',
+            expected_exceptions=waiting.exceptions.TimeoutExpired
+        )
+
+
+def deploy_default_secret(
+        kube_api_client: ApiClient,
+        name: str,
+        ignore_conflict: bool = True,
+        pull_secret: Optional[str] = None
+) -> Secret:
+    if pull_secret is None:
+        pull_secret = os.environ.get('PULL_SECRET', '')
```

Comment:
learning from my mistakes, we should validate this thing is a valid JSON as soon as possible
Can you do a ``json.loads`` for this validation?